src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -66,20 +67,32 @@ object ClusterStatus extends Enumeration {
 
 
 object Cluster {
-  def create(clusterRequest: ClusterRequest, clusterName: ClusterName, googleProject: GoogleProject, googleId: UUID, operationName: OperationName): Cluster = Cluster(
-    clusterName = clusterName,
-    googleId = googleId,
-    googleProject = googleProject,
-    googleServiceAccount = clusterRequest.serviceAccount,
-    googleBucket = clusterRequest.bucketPath,
-    clusterUrl = getClusterUrl(googleProject, clusterName),
-    operationName = operationName,
-    status = ClusterStatus.Creating,
-    hostIp = None,
-    createdDate = Instant.now(),
-    destroyedDate = None,
-    labels = clusterRequest.labels,
-    jupyterExtensionUri = clusterRequest.jupyterExtensionUri)
+  def create(clusterRequest: ClusterRequest, clusterName: ClusterName, googleProject: GoogleProject, googleId: UUID, operationName: OperationName, clusterDefaultsConfig: ClusterDefaultsConfig): Cluster = {
+    val numWorkers = clusterRequest.numberOfWorkers.getOrElse(clusterDefaultsConfig.numberOfWorkers)
+
+    Cluster(
+        clusterName = clusterName,
+        googleId = googleId,
+        googleProject = googleProject,
+        googleServiceAccount = clusterRequest.serviceAccount,
+        googleBucket = clusterRequest.bucketPath,
+        numberOfWorkers = numWorkers,
+        masterMachineType = clusterRequest.masterMachineType.getOrElse(clusterDefaultsConfig.masterMachineType),
+        masterDiskSize = clusterRequest.masterDiskSize.getOrElse(clusterDefaultsConfig.masterDiskSize),
+        workerMachineType = if (numWorkers == 0) None else Some(clusterRequest.workerMachineType.getOrElse(clusterDefaultsConfig.workerMachineType)),
```

Comment:
can you break this behaviour out into a function? something that looks like fn(maybeZero, desiredValue, fallbackValue)